pkg/inject/sidecar.go

```diff
@@ -132,11 +133,12 @@ func container(jaeger *v1.Jaeger, dep *appsv1.Deployment) corev1.Container {
 	jgBinaryTrft := util.GetPort("--processor.jaeger-binary.server-host-port=", args, 6832)
 
 	if len(util.FindItem("--jaeger.tags=", args)) == 0 {
-		agentTags := fmt.Sprintf("%s=%s,%s=%s,%s=%s,%s=%s",
+		agentTags := fmt.Sprintf("%s=%s,%s=%s,%s=%s,%s=%s,%s=%s",
 			"cluster", "undefined", // this value isn't currently available
 			"deployment.name", dep.Name,
 			"pod.namespace", dep.Namespace,
 			"pod.name", fmt.Sprintf("${%s:}", envVarPodName),
+			"host.ip", fmt.Sprintf("${%s:}", envVarHostIP),
```

Comment:
For #865 this should be the `ip` tag, specifically so that the Clock Skew Adjustor would recognise [different spans from the same host as the same](https://github.com/jaegertracing/jaeger/blob/master/model/adjuster/clockskew.go#L78), and not completely mess-up the spans if two k8s nodes were out of clock-sync.

Reply:
changed.

Reply:
This needs confirmation, but I don't think that's how it works. The `ip` tag that you see is under `process` and is gathered by the tracer and I don't think we have any influence on that in most cases...

Reply:
According to [Agent level tags](https://www.jaegertracing.io/docs/1.16/deployment/#agent-level-tags), tags passed to `--jaeger.tags` are
> added to the *process* tags of all spans passing through the agent.
(Emphasis mine)

Reply:
Given
```yaml
apiVersion: jaegertracing.io/v1
kind: "Jaeger"
metadata:
  name: "hampy-jaeger"
  namespace: "observability"
spec:
  strategy: allInOne
  ingress:
    hosts:
      - hampy-jaeger.dev.kepler.wgsyd
  agent:
    options:
        jaeger.tags: injected_agent_tag=true
```

![image](https://user-images.githubusercontent.com/138397/73040248-fd1c2a80-3eac-11ea-9024-6eee61f2e3a1.png)

Note: The `ip` process tag is set in our code, to thwart the Clock Skew Adjustor, pending this change.

Reply:
Decided to give it a try myself, and confirmed that this isn't going to work:

![image](https://user-images.githubusercontent.com/13387/73056189-b28bb580-3e8e-11ea-8df7-4ecfe1cf466f.png)

Note how the `ip` tag is duplicated, as the `ip` tag is added by the client automatically. We had a similar case of duplicate tags in the collector, but we can't de-dup it there, as the tags have different values, we can't know which one is right.

IMO, if we want to have the `hostIP` for a given span, its name should be qualified so that it's not ambiguous. Something like @rubenvp8510 did before, `host.ip`. This won't fix the clock skew problem, but neither will the current state of this PR.

Note that I had to apply the fix from #864 to get this tested.

Reply:
Drat. I had implied from the docs that the values would be overwritten by the agent, but did not test that flow in my earlier test.

I don't have this issue here because I'm using OpenCensus, which does not add the `ip` tag automatically.

`host.ip` is consistent, but sadly doesn't resolve #865, which is perhaps not-doable, as demonstrated.

It's a shame because the Client library is providing the wrong IP for the Clock Skew Adjustor's algorithm.

I guess the best hope of progress would be if the Clock Skew Adjustor supported a configurable tag. Then I could specify `host.ip` knowing it was always available in agent-injected pods.

Reply:
As consumer of the client libraries, what you can do is initialize the tracer [with a set of tags](https://github.com/jaegertracing/jaeger-client-java/blob/74adce4c4b4c507ccfb010ffb06c95973cc97c6a/jaeger-core/src/main/java/io/jaegertracing/internal/JaegerTracer.java#L653-L658), which might include the IP with the appropriate hostIP. If that tag is already set by the time the tracer is initialized, that value will be used instead:

https://github.com/jaegertracing/jaeger-client-java/blob/74adce4c4b4c507ccfb010ffb06c95973cc97c6a/jaeger-core/src/main/java/io/jaegertracing/internal/JaegerTracer.java#L109-L125

Reply:
That's what I'm doing now, visible in my screenshot above. The effort is simply that we aren't otherwise exposing the Host IP to the pods, so we'll have to plumb that through into each pod to pass into the trace initialisation, which means a half-dozen two-file updates to various services.

Thinking about it, since in my case I don't have the conflict over the `ip` tag, once this change goes in I could put in my Jaeger something like
```yaml
spec:
  agent:
    options:
        jaeger.tags: ip=${HOST_IP:}
```

although because the jaeger.tags option's presence hides all the auto-added tags, I then lose the other auto-added process tags, which would be disappointing.

I think I'll stick with the constant string to defang the Clock Skew Adjuster, and my next effort to get the Clock Skew Adjuster doing the right thing will be to get it to honour the new `host.ip` tag instead of an `ip` tag for its machine-key.

Still, even without the Clock Skew Adjuster use-case I opened #865 for, `host.ip` is a valuable process tag to add.